OData/src/System.Web.OData/OData/Query/Expressions/AggregationBinder.cs

```diff
@@ -173,8 +173,13 @@ private IQueryable BindSelect(IQueryable grouping)
 
         private Expression CreateAggregationExpression(ParameterExpression accum, AggregateExpression expression)
         {
-            LambdaExpression propertyLambda = Expression.Lambda(BindAccessor(expression.Expression),
-                this._lambdaParameter);
+            LambdaExpression propertyLambda = null;
+            // $count is a virtual property, so there's not a propertyLambda to create.
+            if (expression.Method != AggregationMethod.VirtualPropertyCount)
```

Comment:
Could you reorder this code to have only one reference to VirtualPropertyCount since it is handled differently?  Something like:

var asQuerableMethod ...
Expression asQuerableExpression ...

if (expression.Method == AggregationMethod.VirtualPropertyCount)
{
    var countMethod = ExpressionHelperMethods.QueryableCountGeneric.MakeGenericMethod(this._elementType); 
    aggregationExpression = Expression.Call(null, countMethod, asQuerableExpression); 
}
else
{
  LambdaExpression propertyLambda = Expression.Lambda(BindAccessor(expression.Expression), this._lambdaParameter); 

  switch (expression.Method)
...


}